tests/x11/window_system.pm

```diff
@@ -0,0 +1,24 @@
+# X11 regression tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Information about current window system
+# Maintainer: Jan Baier <jbaier@suse.cz>
+
+sub run {
+    x11_start_program('xterm');
+    my $window_system = script_output('echo $XDG_SESSION_TYPE');
+    script_run('exit', 0);
+    record_info("$window_system", "Current window system is $window_system");
+}
+
+sub test_flags {
+    return {fatal => 0};
```

Comment:
fatal is not the default so I guess you can simply delete the test_flags section. In case you really want to ignore errors in this module you can use the option [ignore_failure](http://open.qa/docs/#_how_to_write_tests), I would not recommend it in this case though. Opening an xterm and outputting the variable sholud really work :) Btw, how about the following way to write:

```
x11_start_program('xterm');
assert_script_run("echo $XDG_SESSION_TYPE | tee /dev/$serialdev");
script_run('exit', 0);
```

This will look like
https://openqa.opensuse.org/tests/699991#step/zypper_lr/3

Reply:
I was not aware of the defaults (afaik it is not in the documentation) so I was defensive.

My solution with record_info has one advantage: the result x11/wayland will squeeze into the small box and it is visible on the first look :)

Reply:
Actually I agree with @okurz  "we should only do this (group all the desktop regression tests into lib/main_common) if the according scenarios are also executed and working for openSUSE".

And now @yuanren10 is working on migrating SLE openQA test cases to TW (such as poo#23884). It would be better to move those regression tests into lib/main_common after the corresponding work is done.

I saw @baierjan  already made changes to move some regression tests into main_common. I don't think it is suitable for now since those tests are not "real" common for now.

Reply:
@okurz, @GraceWang571 I am not aware of any move I did outside this PR. So let's split this merge request and leave the relocation of the regression tests to later (after @yuanren10).

Reply:
Thanks everyone! I am pretty for it that the splitting the request is the best solution for the x11/wayland exam part to check-in.

About the targets moving to the lib/main_common.pm, there seems quite some cases need to burnish, we'll probably need to break them down into separate p.o.o tickets to have a further track. I'll do it if nobody is against the way to handle them.